### ISSUE TITLE
"ct-ng build.1" hangs, cpu 100%

### ISSUE THREAD

**Issue by antekone** [crosstool-ng/crosstool-ng#1975]
I'm trying to run the `build.1` command, but crosstool-NG hangs during build of libc, and stays on 100% CPU utilization. The same happens with `build`.

ps auxf:

```
buildbot   86188  0.0  0.0  16424  4892 ?        Ss   Jun05   0:17 tmux
buildbot   86189  0.0  0.0  11228  5504 pts/1    Ss   Jun05   0:00  \_ -bash
buildbot  333645  0.0  0.0  14132  3968 pts/1    S+   10:26   0:00  |   \_ /usr/bin/make -rf /home/buildbot/dev/thirdparty/.bin/bin/ct-ng build.1
buildbot  333653  0.0  0.0  14128  4096 pts/1    S+   10:26   0:00  |       \_ /usr/bin/make -rf /home/buildbot/dev/thirdparty/.bin/bin/ct-ng build CT_JOBS=1
buildbot  333661  0.0  0.0  13352  7296 pts/1    S+   10:26   0:01  |           \_ /usr/bin/bash /home/buildbot/dev/thirdparty/.bin/share/crosstool-ng/scripts/crosstool-NG.sh
buildbot  491732  0.0  0.0  13352  5604 pts/1    S+   11:25   0:00  |               \_ /usr/bin/bash /home/buildbot/dev/thirdparty/.bin/share/crosstool-ng/scripts/crosstool-NG.sh
buildbot  493826  0.0  0.0  13352  5480 pts/1    S+   11:25   0:00  |                   \_ /usr/bin/bash /home/buildbot/dev/thirdparty/.bin/share/crosstool-ng/scripts/crosstool-NG.sh
buildbot  493830  0.0  0.0   8272  3200 pts/1    S+   11:25   0:00  |                       \_ make -j1 -l CXX= default_cflags= BUILD_CFLAGS=-O2 -g -I/home/buildbot/dev/thirdparty/crosstool-ng/.build/x86_64-centos7-linux-gnu/buildtools/include   BUILD_CPPFLAGS= BUILD_LDFLAGS=-L/home/buildbot/dev/thirdparty/crosstool-ng/.build/x86_64-centos7-linux-gnu/buildtools/lib   all
buildbot  493832  0.0  0.0  15176  7936 pts/1    S+   11:25   0:00  |                       |   \_ make -r PARALLELMFLAGS= -C /home/buildbot/dev/thirdparty/crosstool-ng/.build/x86_64-centos7-linux-gnu/src/glibc objdir=/home/buildbot/dev/thirdparty/crosstool-ng/.build/x86_64-centos7-linux-gnu/build/build-libc/multilib all
buildbot  513940  0.0  0.0  13248  7808 pts/1    S+   11:32   0:00  |                       |       \_ make subdir=elf -C elf ..=../ subdir_lib
buildbot  514353  0.0  0.0   8524  3200 pts/1    S+   11:32   0:00  |                       |           \_ make -f /home/buildbot/dev/thirdparty/crosstool-ng/.build/x86_64-centos7-linux-gnu/build/build-libc/multilib/elf/librtld.mk -f rtld-Rules
buildbot  514354 84.4  0.0  10828  5504 pts/1    R+   11:32  76:10  |                       |               \_ make subdir=csu -C ../csu ..=../ objdir=/home/buildbot/dev/thirdparty/crosstool-ng/.build/x86_64-centos7-linux-gnu/build/build-libc/multilib -f Makefile -f ../elf/rtld-Rules rtld-all rtld-modules=rtld-check_fds.os rtld-errno.os rtld-init-arch.os
buildbot  493831  0.0  0.0  13352  5480 pts/1    S+   11:25   0:00  |                       \_ /usr/bin/bash /home/buildbot/dev/thirdparty/.bin/share/crosstool-ng/scripts/crosstool-NG.sh
buildbot  493833  0.0  0.0  13352  5352 pts/1    S+   11:25   0:00  |                           \_ /usr/bin/bash /home/buildbot/dev/thirdparty/.bin/share/crosstool-ng/scripts/crosstool-NG.sh
buildbot  493836  0.0  0.0   2656  1152 pts/1    S+   11:25   0:00  |                           |   \_ cat -
buildbot  493835  0.2  0.0  13352  5480 pts/1    S+   11:25   0:14  |                           \_ /usr/bin/bash /home/buildbot/dev/thirdparty/.bin/share/crosstool-ng/scripts/crosstool-NG.sh
```

Any ideas what causes this? The filesystem on this machine is ext4, and it's a virtual machine on VMware ESX (12 GB of RAM -- is this too little?). I left it for 24hours, but the state was the same as when I left it (so "make" took 100% CPU).

Just checked and the same happens on another machine, which is a physical ArchLinux x86_64 machine, the same architecture, the same kernel.

crosstool-NG version 1.25.0.196_227d99d

ArchLinux x86_64: Linux hyperion 6.3.5-arch1-1 #1 SMP PREEMPT_DYNAMIC Tue, 30 May 2023 13:44:01 +0000 x86_64 GNU/Linux

Attached files:

[build.log.gz](https://github.com/crosstool-ng/crosstool-ng/files/11663653/build.log.gz)
[config.gz](https://github.com/crosstool-ng/crosstool-ng/files/11663655/config.gz)

**Comment by antekone** [comment]
This can be a duplicate of https://github.com/crosstool-ng/crosstool-ng/issues/1946, because ArchLinux ships with GNU Make 4.4.1

**Comment by antekone** [comment]
Fixed by downgrading make from 4.4 to 4.3.

I've downloaded make-4.3 sources, compiled it myself, make installed to a chosen prefix directory, added this directory to the beginning of $PATH, removed crosstool-ng sources, re-cloned it again, bootstrapped, configured, it picked up my make-4.3, built the toolchain with `ct-ng build` and it finished successfully.
